src/main/java/io/getunleash/DefaultUnleash.java

```diff
@@ -168,40 +169,89 @@ private FeatureEvaluationResult getFeatureEvaluationResult(
                     fallbackAction.test(toggleName, enhancedContext), defaultVariant);
         } else if (!featureToggle.isEnabled()) {
             return new FeatureEvaluationResult(false, defaultVariant);
-        } else if (featureToggle.getStrategies().size() == 0) {
+        } else if (featureToggle.getStrategies().isEmpty()) {
             return new FeatureEvaluationResult(
                     true, VariantUtil.selectVariant(featureToggle, context, defaultVariant));
         } else {
-            for (ActivationStrategy strategy : featureToggle.getStrategies()) {
-                Strategy configuredStrategy = getStrategy(strategy.getName());
-                if (configuredStrategy == UNKNOWN_STRATEGY) {
-                    LOGGER.warn(
-                            "Unable to find matching strategy for toggle:{} strategy:{}",
-                            toggleName,
-                            strategy.getName());
-                }
+            // Dependent toggles, no point in evaluating child strategies if our dependencies are
+            // not enabled
```

Comment:
not enabled -> not satisfied. satisfied may mean disabled or enabled depending on how we configure our dependencies